lib/MusicBrainz/Server/Edit/Release/AddCoverArt.pm

```diff
@@ -105,11 +105,19 @@ sub build_display_data {
     my ($self, $loaded) = @_;
 
     my $loaded_release = $loaded->{Release}{ $self->data->{entity}{id} };
-    my $release = $loaded_release ||
+
+    # The release used to link (if it still exists) from the release field.
+    my $field_release = $loaded_release ||
         Release->new(
-            id => $self->data->{entity}{id},
             name => $self->data->{entity}{name},
+            # Allow linking to release redirecting to still existing release.
+            $self->c->model('Release')->get_by_gid($self->data->{entity}{mbid}) ? (
```

Comment:
I remembered that the `foreign_keys` method also supports loading entities by MBID, IIRC. So I believe we can just replace the reference to `$self->data->{entity}{id}` in `foreign_keys`, and also in the `$loaded_release` assignment above, to point to the `mbid` property instead (unless I am missing something).